src/tests/ffi.cpp

```diff
@@ -5820,3 +5820,85 @@ TEST_F(rnp_tests, test_ffi_security_profile)
 
     rnp_ffi_destroy(ffi);
 }
+
+static void
+decrypt_hidden_to(rnp_ffi_t &ffi)
+{
+    assert_true(import_sec_keys(ffi, "data/keyrings/1/secring.gpg"));
+    assert_rnp_success(
+      rnp_ffi_set_pass_provider(ffi, ffi_string_password_provider, (void *) "password"));
+
+    rnp_input_t  input = NULL;
+    rnp_output_t output = NULL;
+    assert_rnp_success(
+      rnp_input_from_path(&input, "data/test_messages/message.txt.hidden-to"));
+    assert_non_null(input);
+    assert_rnp_success(rnp_output_to_memory(&output, 0));
+    assert_rnp_success(rnp_decrypt(ffi, input, output));
+    uint8_t *buf = NULL;
+    size_t   len = 0;
+    rnp_output_memory_get_buf(output, &buf, &len, false);
+    std::string out;
+    out.assign((char *) buf, len);
+    assert_true(out == file_to_str("data/test_messages/message.txt"));
+    rnp_input_destroy(input);
+    rnp_output_destroy(output);
+}
+
+TEST_F(rnp_tests, test_ffi_decrypt_hidden_to)
+{
+    rnp_ffi_t ffi = NULL;
+    test_ffi_init(&ffi);
+
+    decrypt_hidden_to(ffi);
+
+    rnp_ffi_destroy(ffi);
+    // TODO tunables to enable/disable trying to decrypt messages hidden recipients
+}
+
+TEST_F(rnp_tests, DISABLED_test_ffi_decrypt_hidden_to_no_errormsgs)
+{
+    // Secret keyrings can have many keys.
+    // When decrypting a message to a hidden recipient, each key has to be tried.
+    // There shouldn't be any error messages printed about decryption failures caused by the
+    // failed attempts.
+    rnp_ffi_t ffi = NULL;
+    assert_rnp_success(rnp_ffi_create(&ffi, "GPG", "GPG"));
+
+    // Log fd handing inspired by test_ffi_set_log_fd.
+    int log_fd = rnp_open("decrypt_hidden_to.log", O_RDWR | O_CREAT | O_TRUNC, 0777);
```

Comment:
It would be easier to use CLI/Python tests to check logging output, like this one:
```
        ret, _, err = run_proc(RNPK, ['--homedir', RNPDIR, '--import-keys', data_path('test_stream_key_load/ecc-25519-pub-4.b64')])
        self.assertEqual(ret, 1)
        self.assertRegex(err, r'(?s)^.*wrong base64 padding: ==zz.*Failed to init/check dearmor.*failed to import key\(s\) from .*, stopping.$')
```